app/models/bookmark.rb

```diff
@@ -1,3 +1,16 @@
 class Bookmark < ActiveRecord::Base
+
   belongs_to :topic
+  before_validation :httpset
+  validates :url, format: { with: /\Ahttp:\/\/.*(com|org|net|gov)/i,
+    message: "only allows valid URLs." }
+
+  def  httpset
+    if self.url =~ /\Ahttp:\/\/|\Ahttps:\/\//i
+    else
+      self.url = "http://"+ self.url
```

Comment:
Nice!

It's a bit more rubyish to use string interpolation like this: `"http://#{self.url}"`. If you decide to keep it the way you have it, add a space before the `+` operator.